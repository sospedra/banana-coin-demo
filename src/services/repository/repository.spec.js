/* global expect, describe, it */
const repositoryService = require('./')

jest.mock('util', () => {
  const fixture = require('./fixture')
  return { promisify: jest.fn(() => () => fixture) }
})

describe('service repository', () => {
  const dirname = 'dirname'
  const dbname = 'dbname'
  const repository = repositoryService(dirname, dbname)

  it('getAll should return all the fixture', async () => {
    const data = await repository.getAll()
    expect(data).toMatchSnapshot()
  })

  it('getByID should return one log from the fixture', async () => {
    const data = await repository.getByID(0)
    expect(data).toMatchSnapshot()
  })

  it('create should return the id of the new log', async () => {
    const data = await repository.create({})
    expect(data).toMatchSnapshot()
  })

  it('update should not fail', async () => {
    await repository.update(0, {})
  })

  it('remove should not fail', async () => {
    await repository.remove(0)
  })

  it('purge should not fail', async () => {
    await repository.purge()
  })
})
