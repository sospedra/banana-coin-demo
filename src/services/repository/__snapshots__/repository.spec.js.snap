// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service repository create should return the id of the new log 1`] = `5`;

exports[`service repository getAll should return all the fixture 1`] = `
Array [
  Object {
    "id": 0,
    "name": "bna",
    "status": "close",
    "value": 2000,
  },
  Object {
    "id": 1,
    "name": "bna",
    "status": "open",
    "value": 1990,
  },
  Object {
    "id": 2,
    "name": "bna",
    "status": "open",
    "value": 1990,
  },
  Object {
    "id": 4,
    "name": "bna",
    "status": "open",
    "value": 1990,
  },
]
`;

exports[`service repository getByID should return one log from the fixture 1`] = `
Object {
  "id": 0,
  "name": "bna",
  "status": "close",
  "value": 2000,
}
`;
